docs/tutorials/gradient-checkpointing.md

```diff
@@ -0,0 +1,488 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.15.2
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(gradient-checkpointing)=
+## Gradient checkpointing with `jax.checkpoint` (`jax.remat`)
+
+In this tutorial, you will learn how to control JAX automatic differentiation's saved values using {func}`jax.checkpoint` (also known as {func}`jax.remat`), which can be particularly helpful in machine learning.
+
+If you are new to automatic differentiation (autodiff) or need to refresh your memory, JAX has {ref}`automatic-differentiation` and {ref}`advanced-autodiff` tutorials.
+
+**TL;DR** Use the {func}`jax.checkpoint` decorator (aliased as {func}`jax.remat`) with {func}`jax.grad` to control which intermediates are saved on the forward pass versus the recomputed intermediates on the backward pass, trading off memory and FLOPs.
+
+**Note:** Don't miss the {ref}`gradient-checkpointing-practical-notes` for a discussion about how {func}`jax.checkpoint` interacts with {func}`jax.jit`.
+
+Without using {func}`jax.checkpoint`, the forward pass of `jax.grad(f)(x)` saves (to use on the backward pass, the values of Jacobian coefficients and other intermediates. These saved values are referred to as _residuals_:
```

Comment:
Something is strange in this sentence. Copy-paste error?

Reply:
`)` instead of `,`. Also rephrasing to make it clearer.